test/Parser/Parser.cpp

```diff
@@ -11,32 +11,340 @@
     glu::Scanner scanner(buf.get());              \
     glu::Parser parser(scanner /*, 1*/)
 
+#define PREP_MAIN_PARSER(str)                                       \
+    std::string wrapped = std::string("func main() {") + str + "}"; \
+    std::unique_ptr<llvm::MemoryBuffer> buf(                        \
+        llvm::MemoryBuffer::getMemBufferCopy(wrapped)               \
+    );                                                              \
```

Comment:
`str` is a string literal, you can just concat them
```suggestion
    std::unique_ptr<llvm::MemoryBuffer> buf(                          \
        llvm::MemoryBuffer::getMemBufferCopy("func main() {" str "}") \
    );                                                                \
```